Aspects/MethodInterceptor.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Castle.DynamicProxy;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Text.Json;
+
+namespace Karma.Aspects
+{
+    public class MethodInterceptor : IInterceptor
+    {
+        private readonly ILogger _logger;
+        public MethodInterceptor(ILogger<MethodInterceptor> logger)
+        {
+            _logger = logger;
+        }
+        public void Intercept(IInvocation invocation)
+        {
+            try
+            {
+                invocation.Proceed();
+
+                _logger.LogInformation($"{invocation.Method.Name} " +
+                    $"called with parameters: {JsonSerializer.Serialize(invocation.Arguments)} " +
+                    $"returned this response: {JsonSerializer.Serialize(invocation.ReturnValue)}");
```

Comment:
You could use a single string here instead of adding three as you are already using formatting.

Reply:
I've changed it to a single string, but the seperation to three is used so you could see the whole string in one standard window of the editor with the most common resolution(1920x1080).

Reply:
Well, this was only a suggestion, you are right, it is way easier to read this way 😄.